src/resolve/resolved-conf.c

```diff
@@ -376,6 +379,71 @@ int config_parse_dnssd_txt(const char *unit, const char *filename, unsigned line
         return 0;
 }
 
+int config_parse_dns_stub_listener_address(const char *unit,
+                                           const char *filename,
+                                           unsigned line,
+                                           const char *section,
+                                           unsigned section_line,
+                                           const char *lvalue,
+                                           int ltype,
+                                           const char *rvalue,
+                                           void *data,
+                                           void *userdata) {
+
+        Manager *m = userdata;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = inet_pton(AF_INET, rvalue, &m->dns_stub_listener_sa.sockaddr.in.sin_addr);
+        if (r <= 0) {
+                log_syntax(unit, LOG_ERR, filename, line, 0, "Failed to parse DNS stub listener address value in '%s', ignoring: %m", rvalue);
+                return 0;
+        }
+
+        return 0;
+}
+
+int config_parse_dns_stub_listener_bindtodevice(
+                const char* unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Manager *m = userdata;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue) || streq(rvalue, "*")) {
+                m->bind_to_device = -1;
+                return 0;
+        }
+
+        r = if_nametoindex(rvalue);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, 0, "Invalid stub listener interface name, ignoring: %s", rvalue);
+                return 0;
+        }
```

Comment:
Hmm, this means that the interface must be already present when resolved parses the config. I guess that's OK.